frontend/src/features/biobuddy/BioBuddyChat.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+  <q-drawer show-if-above
+                :width="drawerWidth"
+                    >
+    <q-list>
+      <q-item-label header>BioBuddy Chat</q-item-label>
+      <q-separator />
+      <q-item v-for="(message, index) in messages" :key="index">
+        <q-item-section>
+          <div v-if="message.type === 'function'" class="function-message">
+            <q-item-label class="text-h7 q-mb-sm">
+              <div class="q-pb-sm q-pt-sm text-bold">
+                <img src="/Biobuddy_icon.svg" class="custom-icon" />
+                {{ displayName(message.role) }}</div>
+              <p> <q-icon name="check_circle" color="purple"></q-icon>
+                <span class="text-h7 text-purple q-ml-sm"> {{ displayContent(message) }} </span>
+              </p>
+              <ul>
+                Params:
+                <li v-for="(param, index) in message.message.parameters" :key="index">
+                  {{ param.name }}: {{ param.value }}
+                </li>
+              </ul>
+            </q-item-label>
+          </div>
+          <div v-else>
+            <q-item-label class="text-h7 q-mb-sm">
+              <div class="q-pb-sm q-pt-sm text-bold">{{ displayName(message.role) }}</div>
+              <p>{{ displayContent(message) }}</p>
+            </q-item-label>
+          </div>
+        </q-item-section>
+      </q-item>
+    </q-list>
+    <div class="q-pa-md">
+      <q-input v-model="newMessage" label="Type a message..." dense filled @keyup.enter="sendMessage">
```

Comment:
<q-input v-model="newMessage" :rules="[val => !!val || 'Message is required']" label="Type a message..." dense filled @keyup.enter="sendMessage">
```suggestion
      <q-input v-model="newMessage" label="Type a message..." dense filled @keyup.enter="sendMessage">
```